libcfnet/client_protocol.c

```diff
@@ -196,7 +197,9 @@ static bool SetSessionKey(AgentConnection *conn)
         return false;
     }
 
-    conn->session_key = (unsigned char *) bp->d;
+    buf = malloc(BN_num_bytes(bp));
+    BN_bn2bin(bp, buf);
+    conn->session_key = buf;
```

Comment:
where is `conn->session_key` freed?

Reply:
It is freed in `DeleteConn()` apparently. But in this function, there is a missing `BN_clear_free()` to avoid the leak of `bp`. But I believe this was also leaking before.